PolyEngine/Engine/Src/Engine.cpp

```diff
@@ -83,11 +90,47 @@ void Poly::Engine::Init(std::unique_ptr<IGame> game, std::unique_ptr<IRenderingD
 
 	// Init game
 	Game->Init();
+
+	if (Editor)
+		Editor->Init();
+}
+
+void Engine::Restart()
+{
+	if (Editor)
```

Comment:
Let's create two separate methods: `StartGame` and `EndGame` which initizlize/deinitialize editor, scene and game. After that remove `Restart` and simply use `EndGame(); StartGame();` to restart. Please also rename Editor Deinit and Init to `OnGameCLose` and `OnGameStart`